modules/programs/awscli.nix

```
@@ -57,13 +64,21 @@ in {
   config = lib.mkIf cfg.enable {
     home.packages = [ cfg.package ];
 
-    home.file."${config.home.homeDirectory}/.aws/config" =
+    home.sessionVariables =
+      (lib.mkIf (cfg.settings.path != "") {
+        AWS_CONFIG_FILE = cfg.settings.path;
+      }) //
+      (lib.mkIf (cfg.credentials.path != "") {
+        AWS_SHARED_CREDENTIALS_FILE = cfg.credentials.path;
+      });
```

Use `mkMerge` when combining, otherwise it mostly works by accident 🙂

```suggestion
    home.sessionVariables = mkMerge [
      (lib.mkIf (cfg.settings.path != "") {
        AWS_CONFIG_FILE = cfg.settings.path;
      })
      (lib.mkIf (cfg.credentials.path != "") {
        AWS_SHARED_CREDENTIALS_FILE = cfg.credentials.path;
      })
    ];
```